### DISCUSSION TITLE
Plotting pole figures for different phases from grain orientation files

### DISCUSSION THREAD

**Discussion by rnpratoori** [mtex-toolbox/mtex#1394] [Ask Anything] [answered]
Hello team,

I am new to plotting pole figures. I am currently trying to plot pole figures for 2 different phases. What I have are the grain orientations and the volume fraction of each phase for each grain. Is there a way to plot pole figures from this data?

In specific - I am working with Si I (cubic) and Si II (tetragonal) phases. No grain is completely belonging to either phase. The volume fraction of Si I can vary from 0.02 to 0.5.

Thanks in advance,
Raghunandan.

**Comment by kilir** [comment, marked as answer]
Hi,
trying to understand your data. Do you have the absolute size for each grain or do they all have roughly the same size? In the latter case, I think you can just assign weights for each orientation according to the volume fractions.
Cheers,
Rüdiger

**Reply by rnpratoori** [reply]
Thanks for your reply!

The grains are almost of the same size - probably a 5% variation which I can ignore for now. Can you elaborate on how I can include the weights in my code or direct me to some material where I can learn about this? Please forgive my ignorance, I am completely new to this field.

Also, in case I have a very high variation in my grain sizes, can I just use a new set of weights that includes the volume fraction for each orientation and volume fraction for the size?

Thanks,
Raghunandan.

**Reply by kilir** [reply]
Hi,
I£m not entire sure how your data looks like , but from your description and assuming that the grains are all the same size, your data comes as Euler angles of each grain (apparently the same? if no just use the orientations directly):
```
% generate some data for demonstrating purposes from what I understand you have one orientation per grain which however is in part cubic and in part tetragonal
rot = rotation.rand(1000);

cs1= crystalSymmetry('cubic')
cs2= crystalSymmetry('tetragonal',[1 1 2])

% some random weights from 0-1
weights_cubic = rand(length(rot),1);
weights_tetragonal = 1- weights_cubic;


odf_cubic = calcDensity(orientation(rot,cs1),'weights',weights_cubic)
odf_tetragonal = calcDensity(orientation(rot,cs2),'weights',weights_tetragonal)

plotPDF(odf_cubic,Miller({1,0,0},{1,1,1},{1,1,0},cs1))
% ...
```

And yes, in case the  sizes of your grains vary, you should probably weight the weights by the grain sizes.

Cheers,
Rüdiger

**Reply by rnpratoori** [reply]
Thanks again for your time! This completely addresses my issue.
